gnome-initial-setup/gis-driver.c

```diff
@@ -373,6 +376,31 @@ gis_driver_locale_changed (GisDriver *driver)
   g_signal_emit (G_OBJECT (driver), signals[LOCALE_CHANGED], 0);
 }
 
+static void
+load_vendor_conf_file (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GKeyFile) vendor_conf_file = g_key_file_new ();
+
+  if(!g_key_file_load_from_file (vendor_conf_file, VENDOR_CONF_FILE,
+                                 G_KEY_FILE_NONE, &error))
+    {
+      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_debug ("Could not read file %s: %s", VENDOR_CONF_FILE, error->message);
```

Comment:
I just noticed on a second reading that the old code this replaces was:

```c
    if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
      g_warning ("Could not read file %s: %s", VENDOR_CONF_FILE, error->message);
```

a. was the old GError domain/code wrong?
b. should this not still be a warning?

Reply:
I don't know if it was wrong, but I've always used `G_IO_ERROR_NOT_FOUND` for such cases, and I have just double checked that it's working as expected.
I will update the debug message to a warning though.

Reply:
The documentation for `g_key_file_load_from_file` says:

> This function will never return a G_KEY_FILE_ERROR_NOT_FOUND error. If the file is not found, G_FILE_ERROR_NOENT is returned.

And I have confirmed that the error returned is G_FILE_ERROR, G_FILE_ERROR_NOT_FOUND:

```python
In [1]: from gi.repository import GLib
In [4]: GLib.KeyFile().load_from_file('/does/not/exist', GLib.KeyFileFlags.NONE)
---------------------------------------------------------------------------
Error                                     Traceback (most recent call last)
<ipython-input-4-865cae36c2e6> in <module>()
----> 1 GLib.KeyFile().load_from_file('/does/not/exist', GLib.KeyFileFlags.NONE)

Error: g-file-error-quark: No such file or directory (4)
```

So the old code was correct and this is wrong

Reply:
Shit. You're actually right and I was testing it the wrong way. Sorry for the noise and thanks for not letting this pass!